lib/atlas/input_helper.rb

```diff
@@ -0,0 +1,20 @@
+module Atlas
+  module InputHelper
+    extend ActiveSupport::Concern
+
+    module ClassMethods
+      # Public: Creates a hash where each key is the name of a share group, and
+      # each value an array containing the inputs belonging to the group.
+      #
+      # Inputs which do not belong to a share group are not included.
+      #
+      # Returns a hash.
+      def by_share_group
+        all.select(&:share_group).each_with_object({}) do |input, groups|
```

Comment:
Ruby has this [built-in](http://www.rubydoc.info/stdlib/core/Enumerable#group_by-instance_method) nowadays :wink:
```
all.select(&:share_group).group_by(&:share_group)
```